enclave/src/main/java/org/hyperledger/besu/enclave/VertxRequestTransmitter.java

```diff
@@ -71,15 +71,14 @@ public VertxRequestTransmitter(final HttpClient httpClient) {
         request.end();
       }
       return result.get();
-    } catch (final ExecutionException e) {
-      if (e.getCause() instanceof EnclaveException) {
-        throw (EnclaveException) e.getCause();
+    } catch (final ExecutionException | InterruptedException e) {
+      if (e.getCause() instanceof EnclaveClientException) {
+        throw (EnclaveClientException) e.getCause();
       }
-      throw new EnclaveException("Failure during reception", e);
-    } catch (final InterruptedException e) {
-      throw new EnclaveException("Task interrupted while waiting for Enclave response.", e);
-    } catch (final Exception e) {
-      throw new EnclaveException("Other error", e);
+      else if (e.getCause() instanceof EnclaveServerException) {
+        throw (EnclaveServerException) e.getCause();
+      }
+      throw new EnclaveIOException("Enclave Communication Failed", e);
```

Comment:
NIT: can we encapsulate this logic into a method `mapToEnclaveException(e)` or something similar? I think it will improve readability. What do you think?

```
[..]
} catch (final ExecutionException | InterruptedException e) {
  throw mapToEnclaveException(e);
}
```